app/domain/authentication/authn_gce/validate_status.rb

```diff
@@ -0,0 +1,24 @@
+module Authentication
+  module AuthnGce
+
+    ValidateStatus = CommandClass.new(
+      dependencies: {
+        discover_identity_provider:  Authentication::OAuth::DiscoverIdentityProvider.new
+      },
+      inputs:       %i()
+    ) do
+
+      def call
+        validate_provider_is_responsive
+      end
+
+      private
+
+      def validate_provider_is_responsive
+        @discover_identity_provider.(
+          provider_uri: PROVIDER_URI
+        )
+      end
+    end
+  end
+end
```

Comment:
Fly by comment: This class seems unnecessary.  Just use the `discovery_identity_provider` directly where needed.

Reply:
this file is needed because it handles the specific status validation. It. is called from the [GCE Authenticator class](https://github.com/cyberark/conjur/blob/gcp-feature-branch/app/domain/authentication/authn_gce/authenticator.rb#L47) that is called from the generic [`validate_status`](https://github.com/cyberark/conjur/blob/gcp-feature-branch/app/domain/authentication/validate_status.rb#L70).

one can argue that the `status` method in the GCE Authenticator class will be defined as:
```
      def status(authenticator_status_input:)
        Authentication::OAuth::DiscoverIdentityProvider.new.(
          provider_uri: PROVIDER_URI
        )
      end
```

but i think that it's better to call the `validate_status` class and handle the logic there, for clarity. It is also open for addition in the future this way and it conforms with the other status validation classes.

Reply:
Yeah, I'm arguing that the method is the better solution in this case.  The class isn't being re-used, and I don't think is adding any clarity beyond the method.  It's actually less clear in that it requires reading a lot more boilerplate before you realize all the content is just in `alidate_provider_is_responsive`.

Reply:
ok, i can live with that.

Reply:
huh but what about specs? it's much easier to write the spec for the command class. or do you think that we don't need to write a spec for the `status` method of this class?
```
class Authenticator
      # This delegates to all the work to the call method created automatically
      # by CommandClass
      #
      # This is needed because we need `valid?` to exist on the Authenticator
      # class, but that class contains only a metaprogramming generated
      # `call(authenticator_input:)` method.  The methods we define in the
      # block passed to `CommandClass` exist only on the private internal
      # `Call` objects created each time `call` is run.
      def valid?(input)
        call(authenticator_input: input)
      end

      def status(authenticator_status_input:)
        Authentication::OAuth::DiscoverIdentityProvider.new.call(
          provider_uri: PROVIDER_URI
        )
      end
    end
```

Reply:
Presumably `Authentication::OAuth::DiscoverIdentityProvider` already has tests, so you should be good to go there.

As for the class where you're using it inside the proposed `validate_provider_is_responsive` method, you'd still be injecting `Authentication::OAuth::DiscoverIdentityProvider` as a dependency, so you'll be able to control that in your unit tests, which should be all you need.

Please lmk if I'm missing anything.  I just made a quick review so far, not a thorough one.  Will try to review in depth later if I have time.

Reply:
i don't follow. are you saying that the code above can use a dependency injection?

Reply:
Currently you inject `discover_identity_provider:  Authentication::OAuth::DiscoverIdentityProvider.new` into `ValidateStatus`.  I'm proposing remove `ValidateStatus` and instead inject `discover_identity_provider:  Authentication::OAuth::DiscoverIdentityProvider.new` into whatever class is currently being injected with `ValidateStatus`.

Reply:
no class is being injected with it. it is used as is in the authenticator class. same as with [`authn-azure`](https://github.com/cyberark/conjur/blob/master/app/domain/authentication/authn_azure/authenticator.rb#L91) and [`authn-oidc`](https://github.com/cyberark/conjur/blob/master/app/domain/authentication/authn_oidc/authenticator.rb#L21)

Reply:
i still think that we should leave the `ValidateStatus` command class for uniformity with the other authenticators. this will be the case for each authenticator that will implement the status validation and i think the logic should go in the command class, however small it is.

Reply:
I will try to look at this more closely later and actually pull down your branch to explore.